### PR TITLE
HOCS-4268 Improve logging for user already exists error

### PR DESCRIPTION
**Pull request by robertdeniszczyc2**

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/info/security/KeycloakService.java [hunk 1]
```diff
@@ -54,23 +55,29 @@ public KeycloakService(
...
+
         if (response.getStatus() != HttpStatus.SC_CREATED) {
             String errorMessage = extractCreateUserErrorMessage(response.getEntity());
-            log.error("Failed to create user {}, status {}", createUserDto.getEmail(), response.getStatus(), value(EVENT, KEYCLOAK_FAILURE));
+            log.error("Failed to create user, status {}", response.getStatus(), value(EVENT, KEYCLOAK_FAILURE));
             throw new KeycloakException(errorMessage, response.getStatus());
         }
 
         String userId = CreatedResponseUtil.getCreatedId(response);
+        log.info("Event: {}. Created new user {}", LogEvent.CREATE_USER_SUCCESS, userId);
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/info/security/KeycloakService.java:79]: Can this be altered - IntelliJ will complain but allows it to be picked up by Kibana.

```suggestion
        log.info("Created new user {}", userId, value(EVENT, LogEvent.CREATE_USER_SUCCESS));
```

File: src/main/java/uk/gov/digital/ho/hocs/info/security/KeycloakService.java [hunk 2]
```diff
@@ -54,23 +55,29 @@ public KeycloakService(
...
         UsersResource usersResource = keycloakClient.realm(hocsRealmName).users();
-
-        if (usersResource.search(createUserDto.getEmail()).size() > 0) {
-            log.warn("User {} creation failed - email already exists", createUserDto.getEmail(),
-                    value(EVENT, KEYCLOAK_FAILURE));
-            throw new ApplicationExceptions.UserAlreadyExistsException();
+        List<UserRepresentation> existingUserReps = usersResource.search(createUserDto.getEmail());
+
+        if (!existingUserReps.isEmpty()) {
+            log.info("Event: {}. Status: {}. User with email already exists as user {}",
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/info/security/KeycloakService.java:61]: Same as the other comment please

File: src/main/java/uk/gov/digital/ho/hocs/info/domain/exception/RestResponseEntityExceptionHandler.java [hunk 3]
```diff
@@ -40,18 +40,18 @@ public ResponseEntity handle(ApplicationExceptions.EntityCreationException e) {
...
-
     @ExceptionHandler(ApplicationExceptions.EntityAlreadyExistsException.class)
     public ResponseEntity handle(ApplicationExceptions.EntityAlreadyExistsException e) {
         log.error("ApplicationExceptions.EntityAlreadyExistsException: {}, Event: {}", e.getMessage(), value(EVENT, e.getEvent()));
         return new ResponseEntity<>(e.getMessage(), CONFLICT);
     }
 
+    @ExceptionHandler(ApplicationExceptions.UserAlreadyExistsException.class)
+    public ResponseEntity handle(ApplicationExceptions.UserAlreadyExistsException e) {
+        log.info("ApplicationExceptions.UserAlreadyExistsException: {}, Event: {}", e.getMessage(), value(EVENT, e.getEvent()));
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/info/domain/exception/RestResponseEntityExceptionHandler.java:51]:
```suggestion
        log.info("ApplicationExceptions.UserAlreadyExistsException: {}", e.getMessage(), value(EVENT, e.getEvent()));
```

**Comment by robertdeniszczyc2** [comment]
Hi @LiamMacP I've now committed the suggestion on this so ready for a review again please

Cheers,
Rob

**Review by LiamMacP — APPROVED** [review]
Changes look good - approved

**Pull request merged**